src/hotspot/share/utilities/bitMap.cpp

```diff
@@ -99,6 +99,54 @@ void GrowableBitMap<BitMapWithAllocator>::resize(idx_t new_size_in_bits, bool cl
   update(map, new_size_in_bits);
 }
 
+template <class BitMapWithAllocator>
+bm_word_t* GrowableBitMap<BitMapWithAllocator>::copy_of_range(idx_t start_bit, idx_t end_bit) {
+  assert(start_bit < end_bit, "End bit must come after start bit");
+  assert(end_bit <= size(), "End bit not in bitmap");
+
+  // We might have extra bits at the end that we don't want to lose
+  idx_t const cutoff = bit_in_word(end_bit);
+  idx_t const start_word = to_words_align_down(start_bit);
+  idx_t const end_word = to_words_align_up(end_bit);
+  bm_word_t* const old_map = map();
+
+  BitMapWithAllocator* derived = static_cast<BitMapWithAllocator*>(this);
+
+  bm_word_t* new_map = derived->allocate(end_word - start_word);
+
+  // All words need to be shifted by this amount
+  idx_t const shift = bit_in_word(start_bit);
+  // Bits shifted out by a word need to be passed into the next
+  idx_t carry = 0;
```

Comment:
`carry` should be of type `bm_word_t` it contains bits of the payload, not an index.